common/lock.c

```diff
@@ -8,254 +8,88 @@
-		}
-		else
-		{
-			info.l_type = F_WRLCK;
-		}
-		info.l_whence = SEEK_SET;
-		info.l_start = 0;
-		info.l_len = 0;
-		info.l_pid = 0;
-
-		if (fcntl(lock->fd, cmd, &info) != -1)
-		{
-			res = 1;
-		}
-	}
-
-	lock->fdrefs += res;
-
-end:
-	return res;
-}
-
-static void tdnflock_release(tdnflock lock)
-{
-	if (!lock || lock->fdrefs <= 1)
-	{
-		return;
-	}
-
-	if (--lock->fdrefs == 1)
-	{
-		struct flock info;
-		info.l_type = F_UNLCK;
-		info.l_whence = SEEK_SET;
-		info.l_start = 0;
-		info.l_len = 0;
-		info.l_pid = 0;
-		(void) fcntl(lock->fd, F_SETLK, &info);
-	}
+    int ret;
+
+    if (IsNullOrEmptyString(lockPath))
+    {
+        pr_err("%s: lockPath is empty\n", __func__);
+        return -1;
+    }
+
+    ret = tdnfCreteLockFile();
+    if (ret < 0)
+    {
+        pr_err("%s: tdnfCreteLockFile failed\n", __func__);
+        return -1;
+    }
+
+    while (1)
+    {
+        if (!flock(gLockFd, LOCK_EX | LOCK_NB))
+        {
+           break;
+        }
+        pr_err("WARNING: failed to acquire lock on: %s, retrying ...\n", lockPath);
+        sleep(1);
+    }
+
+  return 0;
 }
 
-/* External interface */
-tdnflock tdnflockNew(const char *lock_path, const char *descr)
+void tdnfLockFree(void)
 {
-	tdnflock lock = NULL;
-
-	if (IsNullOrEmptyString(lock_path) || IsNullOrEmptyString(descr))
-	{
-		goto end;
-	}
-
-	lock = tdnflock_new(lock_path, descr);
-	if (!lock)
-	{
-		pr_err("can't create %s lock on %s (%s)\n",
-				descr, lock_path, strerror(errno));
-	}
-
-end:
-	return lock;
-}
-
-int tdnflockAcquire(tdnflock lock)
-{
-	int locked = 0; /* assume failure */
-
-	if (!lock)
-	{
-		return locked;
-	}
-
-	locked = tdnflock_acquire(lock, TDNFLOCK_WRITE);
-	if (!locked && (lock->openmode & TDNFLOCK_WRITE))
-	{
-		pr_crit("waiting for %s lock on %s\n", lock->descr, lock->path);
-		locked = tdnflock_acquire(lock, (TDNFLOCK_WRITE|TDNFLOCK_WAIT));
-	}
-
-	if (!locked)
-	{
-		pr_err("can't create %s lock on %s (%s)\n", lock->descr, lock->path,
-													strerror(errno));
-	}
-
-	return locked;
-}
-
-void tdnflockRelease(tdnflock lock)
-{
-	if (lock)
-	{
-		tdnflock_release(lock);
-	}
-}
-
-tdnflock tdnflockNewAcquire(const char *lock_path, const char *descr)
-{
-	tdnflock lock = NULL;
-
-	if (IsNullOrEmptyString(lock_path) || IsNullOrEmptyString(descr))
-	{
-		goto end;
-	}
-
-	lock = tdnflockNew(lock_path, descr);
-
-	if (!tdnflockAcquire(lock))
-	{
-		lock = tdnflockFree(lock);
-	}
-
-end:
-	return lock;
-}
-
-tdnflock tdnflockFree(tdnflock lock)
-{
-	if (lock)
-	{
-		tdnflock_release(lock);
-		tdnflock_free(lock);
-	}
-
-	if (remove(TDNF_INSTANCE_LOCK_FILE))
-	{
-			pr_err("WARNING: Unable to remove lockfile(%s)\n",
-							TDNF_INSTANCE_LOCK_FILE);
-	}
-
-	return NULL;
+    if (gLockFd >= 0)
+    {
+        if (flock(gLockFd, LOCK_UN))
+        {
+          pr_err("ERROR: failed to unlock: '%s'\n", lockPath);
+        }
+        (void) close(gLockFd);
+    }
+
+    if (!access(lockPath, F_OK) && remove(TDNF_INSTANCE_LOCK_FILE))
```

Comment:
Here you use the macro `TDNF_INSTANCE_LOCK_FILE` for `remove()` already, but in `access()` and above you use `lockPath`. No matter what you prefer (and I prefer the macro unless there is a chance it changes), make it consistent.